File: android/app/src/main/java/com/bitportal/wallet/reactnativenavigation/parse/parsers/ColorParser.java
package com.reactnativenavigation.parse.parsers;

import com.reactnativenavigation.parse.params.Colour;
import com.reactnativenavigation.parse.params.NullColor;

import org.json.JSONObject;

public class ColorParser {
    public static Colour parse(JSONObject json, String color) {
        return json.has(color) ? new Colour(json.optInt(color)) : new NullColor();
    }
}

File: ios/bitportal/Navigation/ColorParser.m
#import "ColorParser.h"
#import "NullColor.h"
#import <React/RCTConvert.h>

@implementation ColorParser

+ (Color *)parse:(NSDictionary *)json key:(NSString *)key {
	return json[key] ? [[Color alloc] initWithValue:[RCTConvert UIColor:json[key]]] : [NullColor new];
}

@end
